mininet/node.py

```diff
@@ -839,6 +839,42 @@ def init( cls ):
         cls.inited = True
 
 
+class XIAHost( Host ):
+    "eXpressive Internet Architecture host"
+
+    def __init__( self, name, **kwargs ):
+        Node.__init__( self, name, **kwargs )
+        self.xids = {}  # mapping of configurable inputs to XID types
+
+    def config( self, hid=None, **params):
+        """hid: private key file names for Host Identifiers eg. hid='pk1,pk2'
+           params: parameters for Node.config()"""
+        r = Node.config( self, **params )
+        self.setParam( r, 'setHID', hid=hid )
+        return r
+
+    def setHID( self, hid=None ):
+        "Assign HID(s) to the node."
+        if hid is None:
+            return
+        self.xids[ 'hid' ] = self.addParameters( hid )
+        # Load the required module
+        moduleDeps( add='xia_ppal_hid' )
+        # Performs actual HID configuration
+        for p in self.xids[ 'hid' ]:
+            self.cmd( 'xip hid new', p )
+            self.cmd( 'xip hid addaddr', p )
+
+    def addParameters( self, xids ):
+        """Store the ',' separated xids as a list.
+           xids: string input for XID(s)"""
+        xidlist = []
+        xids = xids.split( ',' )
+        for xid in xids:
+            xidlist.append( xid )
+        return xidlist
```

Comment:
A couple of issues here. I think the code inside this function be replaced simply with:

    return xids.split( ',' )

However, the code doesn't take into account that `xids` could be `None`, in which case I think calling `xids.split( ',' )` will cause an exception.

I think we can avoid this issue altogether by expecting XIDs to be passed in as a list instead of a comma-separated string. In other words, instead of allowing:

    xia1 = net.addHost( 'xia1', cls=XIAHost, hid='foo, bar' )

We would have:

    xia1 = net.addHost( 'xia1', cls=XIAHost, hid=['foo', 'bar'] )

This would allow us to drop the function `addParameters()` completely and use the `config()` function as I mentioned above.

Reply:
Hmm, try printing out `params` inside of `config()`. You should see something like `{ 'hid': ['foo', 'bar'] }` I believe.